metrics/web/docs/features/sentry/01_sentry_design.md

```diff
@@ -0,0 +1,167 @@
+- To create a browser context, we should use a `Browser` class.
+- To create a device context, we should use a `Device` class.
+- To create an operation system context, we should use an `OperationSystem` class.
+- To create a GPU context, we should use a `GPU` class.
+- To create a runtime context, we should use a `Runtime` class.
+
+Example of adding a common context using the Sentry SDK built-in classes:
+
+```dart
+final browser = Browser(name: 'chrome', version: '87.0.4280.88');
+Sentry.configureScope((scope) => scope.setContexts(Browser.type, browser));
+```
+
+## Sentry Util
+
+To provide an easy way to use the Sentry SDK, we should create a wrapper named `SentryUtil`. 
+
+The wrapper should contain methods, that meet our requirements described in the [design steps](#design):
+
+- a `configureDefaultContexts` private method that configures default contextual data on the error event. It should add context about the current user `operation system`, and depending on running the application on the `web` or `mobile` - about the current user `browser` or `device`;
+- an `initialize` method , that simplify [initializing the Sentry](#initializing-the-sentry) with an ability to [capture uncaught Dart and Flutter errors](#capturing-uncaught-flutter-and-dart-errors), set the DSN and release name;
+- a `captureException` method, that delegates to the `Sentry` method described in the [Capturing exceptions](#capturing-exceptions);
+- an `addContext` method, that delegates to the `Sentry` method described in the [Adding Contexts](#adding-contexts).
+
+The following class diagram represents the classes required for Sentry integration: 
+
+![Sentry class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/sentry_design/metrics/web/docs/features/sentry/diagrams/sentry_util_class_diagram.puml)
+
+## Data Source Name (DSN)
+
+The [DSN](https://docs.sentry.io/product/sentry-basics/dsn-explainer/) tells the SDK where to send the events. If this value is not provided, the SDK will try to read it from the `SENTRY_DSN` environment variable. If that variable also does not exist, the SDK will just not send any events.
+
+The `DSN` is not a secret, worst thing someone could do is sending events to your account. If that ever happens we have a few options tackling this, we can either [block off certain request](https://docs.sentry.io/product/accounts/quotas/#inbound-data-filters) or cycle the `DSN`.
+
+Despite the above, we still should to improve our security in using the `DSN`, so we should build the Flutter app providing the `DNS` enviroment variable. Also, the environment variable save us from the hard-coding the `DSN`.
+
+```bash
+export SENTRY_DSN=`your_dsn_here`
+flutter build web --dart-define=SENTRY_DSN=$SENTRY_DSN
+```
+
+Then in the app code we can take the `DSN` using `String.fromEnviroment()` method:
+
+```dart
+const dsn = const String.fromEnvironment('SENTRY_DSN');
+```
+
```

Comment:
```suggestion

### Release

```
I suppose we should provide and information on how we can set the release as well. Consider the following points to cover in this sub-section:
- describe why we need the release to be set;
- notice that for the Android and IOS the Sentry SDK takes the release automatically (https://docs.sentry.io/platforms/flutter/configuration/releases/#bind-the-version); 
- provide examples in what can be a release value for the web application;
- describe defining a variable and using it within the application.

Consider the link: https://docs.sentry.io/platforms/flutter/configuration/releases/